packages/fast-components-react-msft/src/button/button.tsx

```diff
@@ -50,12 +59,34 @@ class Button extends Foundation<IButtonHandledProps & IManagedClasses<IMSFTButto
         }
     }
 
-    private generateInnerContent(): React.ReactElement<HTMLSpanElement> | (React.ReactNode | React.ReactNode[]) {
-        if (this.props.appearance === ButtonAppearance.lightweight || this.props.appearance === ButtonAppearance.justified) {
-            return <span className={get(this.props, "managedClasses.button_span")}>{this.props.children}</span>;
-        }
+    /**
+     * Renders slotted children in the appropriate slot
+     */
+    private renderChildrenBySlot(slot: ButtonSlot): JSX.Element[] {
+        return React.Children.map(this.props.children, (child: JSX.Element, index: number) => {
+            if (child.props && child.props.slot === slot) {
+                return (
+                    <React.Fragment key={index}>
+                        {child}
+                    </React.Fragment>
+                );
+            }
+        });
+    }
+
+    /**
+     * Renders the non-slot child content
+     */
+    private renderContent(): JSX.Element {
+        const content: any[] = [];
+
+        React.Children.forEach(this.props.children, (child: JSX.Element, index: number) => {
```

Comment:
What you're effectively doing here is filtering out items, so the more semantic way to do this is:
```
React.Children.toArray(this.props.children).filter((() => {
//...
})`
```

Doing it this way has two benefits: the method name accurately describes what you're doing and you avoid mutating functions

Reply:
Sure